src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -83,16 +84,31 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
             try
             {
                 bool isOffline = CheckAppOffline();
+                bool hasHostConfigError = false;
 
-                _host = BuildHost(isOffline);
+                try
```

Comment:
Can we avoid the nesting here? It would be good to avoid duplicating the code that sets the exceptions and other flags. 

Perhaps check for that exception type in our existing catch block and conditionally take the recovery action if the exception type is `HostConfigurationException`?